runtimevar/filevar/filevar_test.go

```diff
@@ -82,3 +82,68 @@ func (h *harness) Mutable() bool { return true }
 func TestConformance(t *testing.T) {
 	drivertest.RunConformanceTests(t, newHarness)
 }
+
+// Filevar-specific tests.
+
+func TestNew(t *testing.T) {
+	dir, err := ioutil.TempDir("", "filevar_test-")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	tests := []struct {
+		description string
+		path        string
+		decoder     *runtimevar.Decoder
+		want        string
+		wantErr     bool
+	}{
+		{
+			description: "empty path results in error",
+			decoder:     runtimevar.StringDecoder,
+			wantErr:     true,
+		},
+		{
+			description: "empty decoder results in error",
+			path:        filepath.Join(dir, "foo.txt"),
+			wantErr:     true,
+		},
+		{
+			description: "basic file works",
+			path:        filepath.Join(dir, "foo.txt"),
+			decoder:     runtimevar.StringDecoder,
+			want:        filepath.Join(dir, "foo.txt"),
+		},
+		{
+			description: "relative file works is turned absolute",
```

Comment:
relative path gets cleaned up? I think all working test cases are turned absolute. Actually this is a case testing `filepath.Abs` calling `filepath.Clean`, we might not need this case at all.